tasks/idrac_update.yml

```diff
@@ -0,0 +1,27 @@
+---
+
+    - name: Check if dsu is installed
+      shell: command -v dsu
+      register: output_dsu_commad
+      failed_when: ( output_dsu_commad.stderr | length  > 0 )
+
+    - name: Set is_dsu_installed variable
+      set_fact:
+        is_dsu_installed: "{{ output_dsu_commad.stdout | length  > 0  }}"
+
+    - name: Upgrade the iDRAC if applicable 
+      shell: dsu --category=DR --non-interactive --apply-upgrades
+      register: idrac_update_output
+      failed_when: ( idrac_update_output.stderr | length  > 0 )
+      when: (is_dsu_installed == True)
+
+    - name: Check if the upgrades are applied
+      set_fact:
+        is_dsu_upgraded: "{{ 'Installed successfully' in idrac_update_output.stdout }}"
```

Comment:
I don't think you need this task file at all -- but just for completeness: look at how it's done in the other file - we check the return codes not the output.